presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixSplitManager.java

```diff
@@ -0,0 +1,252 @@
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import static io.prestosql.plugin.phoenix.PhoenixErrorCode.PHOENIX_ERROR;
+import static java.util.Objects.requireNonNull;
+
+public class PhoenixSplitManager
+        implements ConnectorSplitManager
+{
+    private static final Logger log = Logger.get(PhoenixSplitManager.class);
+
+    private final PhoenixClient phoenixClient;
+    private PhoenixMetadata phoenixMetadata;
+    private final Map<String, HostAddress> hostCache = new HashMap<>();
+
+    @Inject
+    public PhoenixSplitManager(PhoenixClient phoenixClient, PhoenixMetadataFactory phoenixMetadataFactory)
+    {
+        this.phoenixClient = requireNonNull(phoenixClient, "client is null");
+        this.phoenixMetadata = phoenixMetadataFactory.create();
+    }
+
+    @Override
+    public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorTableLayoutHandle layout, SplitSchedulingStrategy splitSchedulingStrategy)
+    {
+        PhoenixTableLayoutHandle layoutHandle = (PhoenixTableLayoutHandle) layout;
+        PhoenixTableHandle handle = layoutHandle.getTable();
+        try (PhoenixConnection connection = phoenixClient.getConnection()) {
+            String inputQuery = phoenixClient.buildSql(
+                    connection,
+                    handle.getCatalogName(),
+                    handle.getSchemaName(),
+                    handle.getTableName(),
+                    layoutHandle.getDesiredColumns(),
+                    layoutHandle.getTupleDomain(),
+                    phoenixMetadata.getColumns(handle, false));
+
+            return new FixedSplitSource(getHadoopInputSplits(inputQuery).stream().map(split -> (PhoenixInputSplit) split).map(split -> {
+                List<HostAddress> addresses = getSplitAddresses(split);
+
+                return new PhoenixSplit(
+                        phoenixClient.getConnectorId(),
+                        handle.getCatalogName(),
+                        handle.getSchemaName(),
+                        handle.getTableName(),
+                        layoutHandle.getTupleDomain(),
+                        addresses,
+                        new WrappedPhoenixInputSplit(split));
+            }).collect(Collectors.toList()));
+        }
+        catch (IOException | InterruptedException | SQLException e) {
+            throw new PrestoException(PHOENIX_ERROR, e);
+        }
+    }
+
+    private List<HostAddress> getSplitAddresses(PhoenixInputSplit split)
+    {
+        List<HostAddress> addresses;
+        try {
+            String hostName = split.getLocations()[0];
+            HostAddress address = hostCache.get(hostName);
+            if (address == null) {
+                address = HostAddress.fromString(hostName);
+                hostCache.put(hostName, address);
+            }
+            addresses = ImmutableList.of(address);
+        }
+        catch (Exception e) {
```

Comment:
This should be `RuntimeException` since there are no checked exceptions thrown.

The log message should probably include some information about the split, otherwise it doesn't seem useful. Also consider how often this will be logged. If it's called for each split, the log will be flooded with this message. Is there any action the administrator can take based on this log message?